csrc/preseg_passes/fmin_fmax_promotion.cpp

```diff
@@ -0,0 +1,298 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <preseg_passes/fmin_fmax_promotion.h>
+
+#include <unordered_map>
+#include <vector>
+
+#include <ir/utils.h>
+#include <logical_domain_map.h>
+
+namespace nvfuser::preseg_passes {
+
```

Comment:
In the nvFuser codebase, our convention is to put file-local things in an anonymous namespace. In this case, we can put everything until line 272 into an anonymous namespace.